### PR TITLE
Sinks call DataSink::close instead of operating _closed directly

### LINKED ISSUE #8726: [Bug] DCHECK failed at VExprContext::close
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

dev-1.0.0

### What's Wrong?

DCHECK FAILED due to closed 

### What You Expected?

should not check fail

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dataroaring**
TabletSink::_is_closed is duplicated with DataSink::_closed and
all sinks should call DataSink::close rather than set _closed
directly.

Fix for https://github.com/apache/incubator-doris/issues/8726.

# Proposed changes

Issue Number: close #8726 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/sink/result_sink.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ Status VResultSink::close(RuntimeState* state, Status exec_status) {
             state->fragment_instance_id());
 
     VExpr::close(_output_vexpr_ctxs, state);
-    _closed = true;
+    DataSink::close(state, exec_status);
```
Comment (morningman) [inline be/src/vec/sink/result_sink.cpp:128]:
```suggestion
    return DataSink::close(state, exec_status);
```

Comment (morningman) [reply be/src/vec/sink/result_sink.cpp:128]: There are other places which have same problem. Please fix them too.

Comment (dataroaring) [reply be/src/vec/sink/result_sink.cpp:128]: I will update it, it is strange, actually I fix all of them at night of yesterday. It seems that I made some mistakes when format a patch.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**